apps/src/blockly/addons/contextMenu.js

```diff
@@ -148,21 +153,59 @@ const registerKeyboardNavigation = function () {
     },
     callback: function (scope) {
       const controller = Blockly.navigationController;
-      scope.workspace.keyboardAccessibilityMode
-        ? controller.disable(scope.workspace)
-        : controller.enable(scope.workspace);
+      if (scope.workspace.keyboardAccessibilityMode) {
+        controller.disable(scope.workspace);
+      } else {
+        controller.enable(scope.workspace);
+        Blockly.navigationController.navigation.focusWorkspace(scope.workspace);
+      }
     },
     scopeType: GoogleBlockly.ContextMenuRegistry.ScopeType.WORKSPACE,
     id: 'keyboardNavigation',
-    weight: 11,
+    weight,
   };
   GoogleBlockly.ContextMenuRegistry.registry.register(keyboardNavigationOption);
 };
 
+/**
+ * Change cursor type for keyboard navigation
+ */
+const registerCursor = function (cursorType, weight) {
+  const cursorOption = {
+    displayText: function () {
+      return isCurrentCursor(cursorType)
+        ? `✓ ${msg.usingCursorType({type: cursorType})}`
+        : `${msg.useCursorType({type: cursorType})}`;
+    },
+    preconditionFn: function (scope) {
+      if (
+        !experiments.isEnabled(experiments.KEYBOARD_NAVIGATION) ||
+        !scope.workspace.keyboardAccessibilityMode
+      ) {
+        return MenuOptionStates.HIDDEN;
+      } else if (isCurrentCursor(cursorType)) {
+        return MenuOptionStates.DISABLED;
+      } else {
+        return MenuOptionStates.ENABLED;
+      }
+    },
+    callback: function (scope) {
+      localStorage.setItem(BLOCKLY_CURSOR, cursorType);
+      Blockly.navigationController.cursorType = cursorType;
```

Comment:
`setNewCursor` already handles this line, right?

Reply:
Ah yes, good catch! Will clean up.